channel.go

```diff
@@ -86,6 +93,59 @@ func (c *serialChannel) setup() error {
 	return nil
 }
 
+func (c *serialChannel) wait() error {
+	var event syscall.EpollEvent
+	var events [1]syscall.EpollEvent
+
+	fd := c.serialConn.Fd()
+	if fd <= 0 {
+		return fmt.Errorf("serial port IO closed")
+	}
+
+	epfd, err := syscall.EpollCreate1(syscall.EPOLL_CLOEXEC)
+	if err != nil {
+		return err
+	}
+	defer syscall.Close(epfd)
+
+	event.Events = syscall.EPOLLOUT | syscall.EPOLLET&0xffffffff
+	event.Fd = int32(fd)
+	if err = syscall.EpollCtl(epfd, syscall.EPOLL_CTL_ADD, int(fd), &event); err != nil {
+		return err
+	}
+	defer func() {
+		syscall.EpollCtl(epfd, syscall.EPOLL_CTL_DEL, int(fd), nil)
+	}()
+
+	for {
+		nev, err := syscall.EpollWait(epfd, events[:], -1)
+		if err != nil {
+			return err
+		}
+
+		for i := 0; i < nev; i++ {
+			ev := events[i]
+			if ev.Fd == int32(fd) {
+				agentLog.WithFields("events", ev.Events).Debug("New serial channel event")
+				// Do not handle HUP if there is data available
+				if ev.Events&syscall.EPOLLOUT != 0 {
```

Comment:
commend is `// Do not handle HUP if there is data available` but this is `EPOLLOUT` ?

Reply:
Sorry, comment outdated. I'll update.